zetaclient/chains/interfaces/interfaces.go

```diff
@@ -255,9 +255,7 @@ type TSSSigner interface {
 	SignBatch(ctx context.Context, digests [][]byte, height uint64, nonce uint64, chainID int64) ([][65]byte, error)
 
 	EVMAddress() ethcommon.Address
-
 	EVMAddressList() []ethcommon.Address
-	BTCAddress() string
-	BTCAddressWitnessPubkeyHash() *btcutil.AddressWitnessPubKeyHash
+	BTCAddress(chainID int64) *btcutil.AddressWitnessPubKeyHash
```

Comment:
I think it should be
```suggestion
	BTCAddress(chainID int64) (*btcutil.AddressWitnessPubKeyHash, error)
```

Reply:
Semantically, a `nil` return indicates an error.
Tss instance has already validated this in `ValidateAddresses` when zetaclient initializes: https://github.com/zeta-chain/node/blob/feat-multiple-btc-tss-address-and-more-signet-static-info/zetaclient/tss/tss_signer.go#L430 so we won't see nil return at all.

Reply:
> Semantically, a nil return indicates an error.

I would insist on having an error as a return value. Most go code is written using this way. In our case we can't have a `nil` chain address thus it should returns smth like `(nil, ErrNotFound)`

Reply:
Okay. I can add and check the error on every caller.

Reply:
error return added https://github.com/zeta-chain/node/pull/2907/commits/7c5fbb943924c4cc69fda434a8599661bbaa4aff